### PR TITLE
refactor: convert to Typescript (no proper lint, no types yet)

### PR DESCRIPTION
**Pull request by kyegupov**
- [ ] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?

Converts the project to Typescript.
Most of the tslint rules disabled for now to keep the diff minimal.
No type definitions added yet.

#### Any background context you want to provide?

We are working on modernizing Gradle support. To make sense of the logic we could use some type definitions.

#### What are the relevant tickets?

Motivated by https://snyksec.atlassian.net/browse/BST-508

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-var packageFormatVersion = 'mvn:0.0.1';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as  path from 'path';
+import * as subProcess from './sub-process';
 
-module.exports = {
-  inspect: inspect,
-};
+var packageFormatVersion = 'mvn:0.0.1';
```
Comment (miiila) [inline lib/index.ts]: I wouldn't mind `var` to `const`, despite it wouldn't be pure refactoring then, but YMMV. (valid for the rest as well)

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "compilerOptions": {
+      "outDir": "./dist",
+      "pretty": true,
+      "target": "es5",
+      "lib": [
+        "es5",
+        "es2015.promise",
+        "dom",
```
Comment (miiila) [inline tsconfig.json]: Do we need `dom` here? Do we manipulate it somewhere?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      ],
+      "module": "commonjs",
+      "allowJs": false,
+      "sourceMap": true,
+      "strict": true,
```
Comment (miiila) [inline tsconfig.json]: We can add `declaration` to generate types, but we can also add it once we will really add some types :-)

Comment (miiila) [inline tsconfig.json]: It's also handy to add `importHelpers: true` and install `tslib` package; it makes compiled javascript smaller and more readable (if needed)

File: tsconfig.json (hunk 2 above)
Comment (kyegupov) [reply tsconfig.json]: Nope. Mindlessly copied from a template. Droppin'.

File: lib/index.ts (hunk 1 above)
Comment (kyegupov) [reply lib/index.ts]: Yeah, tslint should prohibit var, but enabling all the lints would pollute the diff, introduce a lot of work and little value. Better deferred to a later PR.

File: tsconfig.json (hunk 3 above)
Comment (kyegupov) [reply tsconfig.json]: Hm, doesn't seem to make any change. But why not.

**Pull request merged**